### PR TITLE
Added initial FAQ

### LINKED ISSUE #54: Support external types
We would like to use an already existing external type in the IDL. In our specific case it's an ID class that belongs to another API and serves to identify objects from that API.

### LINKED ISSUE #70: Generate OO-API and corresponding mappers along with the Service API
I know you will probably close this issue right away, but since the other day, when you explained ServiceIDL's concept of the automatically generated Service API vs the manually defined, local OO-API, this didnt leave my head, so I just want to write it down.

Basically, what we subconciously hoped for when we decided to use ServiceIDL was that we would write an IDL, "push the button" and start working with the Java generated entities on the Java side and the C++ generated entities on the other side. We didn't expect that we had to manually write an OO-API to use locally, and the corresponding mapper that maps the OO-API to the Service-API - on both sides.

I'm sure there are convincing technical reasons for having to to this manually, but from a mere user perspective, if I had a magic wand, I'd still love to be able to use the generated entities right away in my code along with my own entities instead of having to map them first.

### PR DESCRIPTION
**Pull request by sigiesec**
Fixes #54 
Fixes #70

---
This change is [Reviewable](https://reviewable.io/reviews/btc-ag/service-idl/73)

### CODE REVIEW

**Comment by sigiesec** [comment]
@GerrietReents Could you in particular have a look at the answers, and check if they are in line with our shared design considerations?

File: FAQ.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+What is a Service API? What is an OO API?
+=========================================
+
+The Service IDL was developed based on the idea that there are Service APIs and OO APIs are distinct concepts, and that
+there should be no hybrids between these two concepts. Based on that, the Service IDL focuses on Service APIs.
+
+The details of this distinction are beyond the scope of this FAQ, but some essential aspects include that
+* a Service API's parameter and return types are non-opaque DTOs and may contain only externalized references to objects via IDs, URIs etc.
+* an OO API's parameter are opaque and are interface types themselves, which offer methods that return object references
```
Comment (huttenlocher) [inline FAQ.md]: "parameter" must be plural: "parameters"

File: FAQ.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+* OMG IDL
+
+What is a Service API? What is an OO API?
+=========================================
+
+The Service IDL was developed based on the idea that there are Service APIs and OO APIs are distinct concepts, and that
+there should be no hybrids between these two concepts. Based on that, the Service IDL focuses on Service APIs.
+
+The details of this distinction are beyond the scope of this FAQ, but some essential aspects include that
+* a Service API's parameter and return types are non-opaque DTOs and may contain only externalized references to objects via IDs, URIs etc.
```
Comment (huttenlocher) [inline FAQ.md]: "parameter" must be plural: "parameters"

**Comment by sebastianhaberey** [comment]
Great Job :+1:

**Pull request merged**